code/webauthn/contracts/contracts/Account.sol

```diff
@@ -0,0 +1,345 @@
+
+    if (_signature.length != 65) {
+      // Signature is invalid anyway, but we need to proceed with the signature verification as usual
+      // in order for the fee estimation to work correctly
+      _signature = new bytes(65);
+
+      // Making sure that the signatures look like a valid ECDSA signature and are not rejected rightaway
+      // while skipping the main verification process.
+      _signature[64] = bytes1(uint8(27));
+    }
+
+    // extract ECDSA signature
+    uint8 v;
+    bytes32 r;
+    bytes32 s;
+    // Signature loading code
+    // we jump 32 (0x20) as the first slot of bytes contains the length
+    // we jump 65 (0x41) per signature
+    // for v we load 32 bytes ending with v (the first 31 come from s) then apply a mask
+    assembly {
+      r := mload(add(_signature, 0x20))
+      s := mload(add(_signature, 0x40))
+      v := and(mload(add(_signature, 0x41)), 0xff)
+    }
+
+    if (v != 27 && v != 28) {
+      magic = bytes4(0);
+    }
+
+    // EIP-2 still allows signature malleability for ecrecover(). Remove this possibility and make the signature
+    // unique. Appendix F in the Ethereum Yellow paper (https://ethereum.github.io/yellowpaper/paper.pdf), defines
+    // the valid range for s in (301): 0 < s < secp256k1n ÷ 2 + 1, and for v in (302): v ∈ {27, 28}. Most
+    // signatures from current libraries generate a unique signature with an s-value in the lower half order.
+    //
+    // If your library generates malleable signatures, such as s-values in the upper range, calculate a new s-value
+    // with 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141 - s1 and flip v from 27 to 28 or
+    // vice versa. If your library also generates signatures with 0/1 for v instead 27/28, add 27 to v to accept
+    // these malleable signatures as well.
+    if (uint256(s) > 0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A0) {
+      magic = bytes4(0);
+    }
+
+    address recoveredAddress = ecrecover(_hash, v, r, s);
+
+    // Note, that we should abstain from using the require here in order to allow for fee estimation to work
+    if (recoveredAddress != owner) {
+      magic = bytes4(0);
+    }
+    if (recoveredAddress == address(0)) {
+      magic = bytes4(0);
+    }
+  }
+
+  function payForTransaction(
+    bytes32,
+    bytes32,
+    Transaction calldata _transaction
+  ) external payable override onlyBootloader {
+    bool success = _transaction.payToTheBootloader();
+    require(success, 'Failed to pay the fee to the operator');
+  }
+
+  function prepareForPaymaster(
+    bytes32, // _txHash
+    bytes32, // _suggestedSignedHash
+    Transaction calldata _transaction
+  ) external payable override onlyBootloader {
+    _transaction.processPaymasterInput();
+  }
+
+  // WARNING: This function is not safe. Anyone can call it and change the owner of the account.
+  // It is only used for testing purposes
+  function updateR1Owner(bytes memory _r1Owner) external {
```

Comment:
Don't forget to update this function. Easier to just add "onlySelf" modifier and the account will be able to update R1Owner if they decide to.